File: 1876-substrings-of-size-three-with-distinct-characters/1876-substrings-of-size-three-with-distinct-characters.py
class Solution:
    def countGoodSubstrings(self, s: str) -> int:
        output = 0
        output = 1
        output = 0
        
        if len(s) < 3:
            return output
        
        for i in range(len(s) - 2):
            if s[i] != s[i + 1] and s[i + 1] != s[i + 2] and s[i] != s[i + 2]:
                output += 1
                
        
       
        return output

File: 1876-substrings-of-size-three-with-distinct-characters/1876-substrings-of-size-three-with-distinct-characters.java
class Solution {
    public int countGoodSubstrings(String s) {
        /*
         * Algorithms & DS: #Sliding Window
         * Time: O(N), Extra Space: O(1)
         */

        int[] checker = new int[26];
        int ans = 0;
        int l = 0, r = 0;

        while (r < s.length()) {
            int rightValue = s.charAt(r) - 'a';

            checker[rightValue]++;

            while (r - l > 2 || checker[rightValue] >= 2) {
                checker[s.charAt(l) - 'a']--;
                l++;
            }

            if(r - l == 2) 
                ans++;
            
            r++;
        }
        
        return ans;
    }
}